python/cuml/decomposition/pca_mg.pyx

```diff
@@ -0,0 +1,604 @@
+            get_cudf_column_ptr(self.singular_values_)
+        cdef uintptr_t mean_ptr = get_cudf_column_ptr(self.mean_)
+        cdef uintptr_t noise_vars_ptr = \
+            get_cudf_column_ptr(self.noise_variance_)
+        cdef cumlHandle* handle_ = <cumlHandle*><size_t>self.handle.getHandle()
+
+        cdef uintptr_t data
+        cdef uintptr_t trans_data
+        arr_interfaces_trans = []
+
+        if self.dtype == np.float32:
+            data = self._build_dataFloat(arr_interfaces)
+            arr_interfaces_trans = self._build_transData(partsToRanks,
+                                                         rnk,
+                                                         self.n_components,
+                                                         np.float32)
+            trans_data = self._build_dataFloat(arr_interfaces_trans)
+
+            fit_transform(handle_[0],
+                          <RankSizePair**>rankSizePair,
+                          <size_t> n_total_parts,
+                          <floatData_t**> data,
+                          <floatData_t**> trans_data,
+                          <float*> comp_ptr,
+                          <float*> explained_var_ptr,
+                          <float*> explained_var_ratio_ptr,
+                          <float*> singular_vals_ptr,
+                          <float*> mean_ptr,
+                          <float*> noise_vars_ptr,
+                          params,
+                          True)
+        else:
+            data = self._build_dataDouble(arr_interfaces)
+            arr_interfaces_trans = self._build_transData(partsToRanks, 
+                                                         rnk,
+                                                         self.n_components,
+                                                         np.float64)
+            trans_data = self._build_dataDouble(arr_interfaces_trans)
+
+            fit_transform(handle_[0],
+                          <RankSizePair**>rankSizePair,
+                          <size_t> n_total_parts,
+                          <doubleData_t**> data,
+                          <doubleData_t**> trans_data,
+                          <double*> comp_ptr,
+                          <double*> explained_var_ptr,
+                          <double*> explained_var_ratio_ptr,
+                          <double*> singular_vals_ptr,
+                          <double*> mean_ptr,
+                          <double*> noise_vars_ptr,
+                          params,
+                          True)
+
+        self.handle.sync()
+
+        for idx in range(n_total_parts):
+            free(<RankSizePair*>rankSizePair[idx])
+        free(<RankSizePair**>rankSizePair)
+
+        self.components_ = cudf.DataFrame()
+        for i in range(0, params.n_cols):
+            n_c = params.n_components
+            self.components_[str(i)] = self.components_ary[i*n_c:(i+1)*n_c]
+
+        if (isinstance(X, cudf.DataFrame)):
```

Comment:
Why only delete X_m when it is a cuDF DataFrame?

Reply:
That's coming from SG PCA. Should we also update that one?

Reply:
Ok, updated it.